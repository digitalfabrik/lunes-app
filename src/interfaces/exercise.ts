export interface IVocabularyOverviewScreen {
  navigation: any
  route: any
}

export interface IVocabularyTrainerScreen {
  navigation: any
  route: any
}

export interface IAlternativeWordProps {
  alt_word: string
  article: string
}

export interface IDocumentProps {
  id: number
  word: string
  article: string
  document_image: Array<{ id: number; image: string }>
  audio: string
  alternatives: IAlternativeWordProps[]
}

export interface IVocabularyOverviewListItemProps {
  id: number
  word: string
  article: string
  image: string
  audio: string
}

export interface IConfirmationModalProps {
  visible: boolean
  setIsModalVisible: Function
  navigation: any
  extraParams: any
}

export interface IAnswerSectionProps {
  tryLater: () => void
  currentDocumentNumber: number
  setCurrentDocumentNumber: Function
  documents: IDocumentProps[]
  finishExercise: Function
  trainingSet: string
  disciplineTitle: string
}

export interface IFeedbackProps {
  secondAttempt: boolean
  result: string
  document?: IDocumentProps
  input: string
}

export interface IActionsProps {
  tryLater: () => void
  giveUp: () => void
  result: string
  checkEntry: () => void
  getNextWord: () => void
  input: string
  isFinished: boolean
  secondAttempt: boolean
}
