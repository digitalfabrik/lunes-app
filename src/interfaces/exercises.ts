export interface IExercisesScreenProps {
  route: any
  navigation: any
}
