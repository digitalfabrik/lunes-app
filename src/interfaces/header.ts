export interface IHeaderProps {
  top: number | undefined
}
