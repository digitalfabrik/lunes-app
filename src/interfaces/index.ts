import { ReactElement } from 'react'

export interface ILoadingProps {
  children: ReactElement
  isLoading: boolean
}

export interface IButtonProps {
  onPress: () => void
  disabled?: boolean
  children: ReactElement
  theme?: string
  testID?: string
}
export interface IPopoverProps {
  setIsPopoverVisible: (param: boolean) => void
  isVisible: boolean
  children: ReactElement
}
