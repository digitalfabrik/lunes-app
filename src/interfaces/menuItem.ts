import { ReactElement } from 'react'

export interface IMenuItemProps {
  selected: boolean
  onPress: () => void
  icon: string
  title: string
  children: ReactElement
}
