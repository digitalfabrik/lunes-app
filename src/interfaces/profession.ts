export interface IProfessionScreenProps {
  navigation: any
}

export interface IProfessionsProps {
  id: number
  title: string
  description: string
  icon?: string
  total_training_sets: number
}
