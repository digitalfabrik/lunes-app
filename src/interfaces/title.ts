import { ReactElement } from 'react'

export interface ITitleProps {
  children: ReactElement
}
