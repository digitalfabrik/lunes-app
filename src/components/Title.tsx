import React from 'react'
import { View, StyleSheet } from 'react-native'
import { ITitleProps } from '../interfaces/title'

export const styles = StyleSheet.create({
  title: {
    minHeight: 54,
    alignItems: 'center',
    marginBottom: 32
  }
})

const Title = ({ children }: ITitleProps) => <View style={styles.title}>{children}</View>

export default Title
