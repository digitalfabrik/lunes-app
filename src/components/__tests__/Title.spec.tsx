import { Text } from 'react-native'
import React from 'react'
import Title from '../Title'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { ITitleProps } from '../../interfaces/title'

describe('Components', () => {
  describe('Title', () => {
    const defaultTitleProps: ITitleProps = {
      children: <Text>Title</Text>
    }

    it('renders correctly across screens', () => {
      const component = shallow(<Title {...defaultTitleProps} />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('should render title passed to it', () => {
      const titleProps: ITitleProps = {
        ...defaultTitleProps
      }

      const component = shallow(<Title {...titleProps} />)
      expect(component.contains(titleProps.children)).toBe(true)
    })
  })
})
