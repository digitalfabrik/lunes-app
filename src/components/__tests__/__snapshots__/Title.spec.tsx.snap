// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Title renders correctly across screens 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginBottom": 32,
      "minHeight": 54,
    }
  }
>
  <Text>
    Title
  </Text>
</View>
`;
